### PR TITLE
Test dinov2 implementation

### PR DESCRIPTION
**Pull request by stegmuel**
## What has changed and why?

Draft fixes for DINOv2.

## How has it been tested?

Tested by running and comparing metrics.
This PR is not intended to be merged.

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Did you update the documentation?

- [ ] Yes
- [x] Not needed (internal change without effects for user)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -32,7 +32,7 @@ dependencies = [
     "pyarrow>=15.0.0",
     "pytorch_lightning>=2.1",
     "tensorboard>=2.10.0",
-    "torch>=2.1.0,<2.6",        # PyTorch Lightning is not yet compatible with PyTorch 2.6
```
Comment (yutong-xiang-97) [inline pyproject.toml:35]: So does removing this constraint work for you? For me on B200, I still need this constraint to let `make install-dev` work.

File: src/lightly_train/_methods/dinov2/dinov2.py [hunk 2]
```diff
@@ -466,12 +494,12 @@ def training_step_impl(
             student_masks_flat=collated_masks,
             n_masked_patches=n_masked_patches,
             masks_weight=masks_weight,
-        )
```
Comment (yutong-xiang-97) [inline src/lightly_train/_methods/dinov2/dinov2.py:469]: In the original impl. there is a tiny difference between the loss value for logging and backprop: https://github.com/facebookresearch/dinov2/blob/4d83ffd4f4b39df112cef7ed7833de7ebf3a202e/dinov2/train/ssl_meta_arch.py#L317. I tried to follow the original flavor but happy to discuss if we need to adjust this.

File: src/lightly_train/_methods/dinov2/dinov2.py [hunk 3]
```diff
@@ -466,12 +494,12 @@ def training_step_impl(
...
+        ) / n_global_crops
 
         koleo_loss = sum(
             self.koleo_loss(token) for token in student_cls_tokens_global.chunk(2)
-        )  # [G, B, D], only use global views
```
Comment (yutong-xiang-97) [inline src/lightly_train/_methods/dinov2/dinov2.py:473]: Same as above.

File: pyproject.toml (hunk 1 above)
Comment (guarin) [reply pyproject.toml:35]: Please don't allow torch 2.6 as PyTorch Lightning is not yet fully compatible with it. You can always override this constraint with UV when installing the package.
